### PR TITLE
fix: [CDS-38227]: Protected access to templateExpressions

### PR DESCRIPTION
**Pull request by Fernando-Dourado**
## Describe your changes
- Add a catch to unblock the save operation.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33861)

### CODE REVIEW

File: 400-rest/src/main/java/software/wings/service/impl/yaml/handler/workflow/ApprovalStepYamlBuilder.java [hunk 1]
```diff
@@ -49,8 +49,13 @@ public class ApprovalStepYamlBuilder extends StepYamlBuilder {
 
   @Override
   public void validate(ChangeContext<StepYaml> changeContext) {
-    List<HashMap<String, Object>> templateExpressions =
-        (List<HashMap<String, Object>>) changeContext.getYaml().getProperties().get("templateExpressions");
+    List<HashMap<String, Object>> templateExpressions = null;
+    try {
+      templateExpressions =
+              (List<HashMap<String, Object>>) changeContext.getYaml().getProperties().get("templateExpressions");
```
[the remark this answers is not in the document]
Comment (Fernando-Dourado) [reply 400-rest/src/main/java/software/wings/service/impl/yaml/handler/workflow/ApprovalStepYamlBuilder.java]: I really forgot, thanks to remember me.

**Review by lasdox — APPROVED** [review]
> Reviewed 1 of 1 files at r3, 4 of 8 files at r7, 4 of 4 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33861)* status: 1. unresolved discussions.

**Comment by abhinav-harness** [comment]
*[`400-rest/src/main/java/software/wings/service/impl/yaml/handler/workflow/ApprovalStepYamlBuilder.java` line 71 at r9](https://reviewable.io/reviews/harness/harness-core/33861#-N42dpCs-LlBmYDFIyqt:-N42dpCs-LlBmYDFIyqu:bo6uh7c) ([raw file](https://github.com/harness/harness-core/blob/863807b65268ef2be5b8412ba4aaf53ca79e9482/400-rest/src/main/java/software/wings/service/impl/yaml/handler/workflow/ApprovalStepYamlBuilder.java#L71)):*
> ```Java
> 
>   private Object getAndSanitizeTemplateExpressions(StepYaml yaml) {
>     Object value = yaml.getProperties().get(TEMPLATE_EXPRESSIONS);
> ```

what if this is null?

**Review by inderschera — APPROVED** [review]
> Reviewed 1 of 1 files at r3, 1 of 4 files at r8, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33861)* status: 1. unresolved discussions. 

___
*[`bazelrc.common` line 10 at r9](https://reviewable.io/reviews/harness/harness-core/33861#-N42k3JQ-K8cdTokDyL-:-N42k3JQ-K8cdTokDyL0:boxq6jz) ([raw file](https://github.com/harness/harness-core/blob/863807b65268ef2be5b8412ba4aaf53ca79e9482/bazelrc.common#L10)):*
> ```
> 
> build --host_java_toolchain=@bazel_tools//tools/jdk:toolchain_java11
> build --java_toolchain=@bazel_tools//tools/jdk:toolchain_java11
> ```

Is this required?

**Review by Fernando-Dourado — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33861)* status: 1 file(s) to review. 3. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/yaml/handler/workflow/ApprovalStepYamlBuilder.java` line 71 at r9](https://reviewable.io/reviews/harness/harness-core/33861#-N42dpCs-LlBmYDFIyqt:-N42xrAX9uF0fuR3FSJd:bwx5z3f) ([raw file](https://github.com/harness/harness-core/blob/863807b65268ef2be5b8412ba4aaf53ca79e9482/400-rest/src/main/java/software/wings/service/impl/yaml/handler/workflow/ApprovalStepYamlBuilder.java#L71)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

what if this is null?
</blockquote></details>

When the variable used in instanceof evaluation is null, it always results in false.
___
*[`bazelrc.common` line 10 at r9](https://reviewable.io/reviews/harness/harness-core/33861#-N42k3JQ-K8cdTokDyL-:-N42ym-kBh_zq_008QbK:b-5zs1l2) ([raw file](https://github.com/harness/harness-core/blob/863807b65268ef2be5b8412ba4aaf53ca79e9482/bazelrc.common#L10)):*
<details><summary><i>Previously, inderschera (Inderpreet Chera) wrote…</i></summary><blockquote>

Is this required?
</blockquote></details>

It is a complete mistake, I revert that change.

**Comment by Fernando-Dourado** [comment]
trigger ti1

**Review by ershadalim — APPROVED** [review]
Looks good

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33861)* status: 1 file(s) to review. 1. unresolved discussions.

**Review by ershadalim — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33861)* status: 1 file(s) to review.

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 4 of 8 files at r7, 2 of 4 files at r8, 1 of 1 files at r10.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33861)* status: :shipit: complete! ✓

**Pull request merged**